src/app/qgisapp.cpp

```diff
@@ -12344,6 +12335,19 @@ QMenu *QgisApp::createPopupMenu()
   return menu;
 }
 
+void QgisApp::notifyUser( QString title, QString message )
```

Comment:
Could this be implmented in the native system wrapper in https://github.com/qgis/QGIS/pull/3964 ?

Also

`const QString &title`

And maybe rename `notifyUser` to `showSystemNotification` or something similar?